Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -2314,10 +2314,18 @@ void WebPageProxy::dispatchActivityStateChange()
         updateCurrentModifierState();
 
     if ((m_potentiallyChangedActivityStateFlags & ActivityState::IsVisible)) {
-        if (isViewVisible())
+        if (isViewVisible()) {
+            // PeriodicalResponsivenessTimer should be started when view becomes visible.
+            m_process->startPeriodicalResponsivenessTimer();
             viewIsBecomingVisible();
-        else
+        } else {
+            // PeriodicalResponsivenessTimer should be stopped when playback is not playing and view is hidden.
+            const MediaProducerMediaStateFlags playingMediaMask { MediaProducerMediaState::IsPlayingAudio, MediaProducerMediaState::IsPlayingVideo };
+            if (!(m_mediaState & playingMediaMask))
+                m_process->stopPeriodicalResponsivenessTimer();
+
```

Comment:
What if other views in the same process are still visible and/or playing media? We should not start/stop the timer from here, we should just notify the process proxy that this particular view changed that state (the visible state is already done with pageIsBecomingInvisible) and the process proxy should start/stop the timer based on all views states.